cmd/util/ledger/migrations/cadence.go

```diff
@@ -345,7 +348,71 @@ func (m *IssueStorageCapConMigration) Close() error {
 	return nil
 }
 
-var _ AccountBasedMigration = &IssueStorageCapConMigration{}
+func (m *IssueStorageCapConMigration) issueAccountCapabilities(
+	migrationRuntime *InterpreterMigrationRuntime,
+	reporter capcons.StorageCapabilityMigrationReporter,
+	address common.Address,
+	capabilities *capcons.AccountCapabilities,
+	handler stdlib.CapabilityControllerIssueHandler,
+) {
+
+	storageMap := migrationRuntime.Storage.GetStorageMap(
+		address,
+		common.PathDomainStorage.Identifier(),
+		false,
+	)
+
+	inter := migrationRuntime.Interpreter
+
+	for _, capability := range capabilities.Capabilities {
+		addressPath := interpreter.AddressPath{
+			Address: address,
+			Path:    capability.Path,
+		}
+
+		borrowStaticType := capability.BorrowType
+
+		if borrowStaticType == nil {
+			reporter.MissingBorrowType(address, addressPath)
+
+			// If the borrow type is missing, treat it as `AnyStruct` or `AnyResource`.
+			// To determine whether it is a resource, read the target path.
```

Comment:
The capability has direct access to the storage path, without any borrow type "restriction", so theoretically the capability could be borrowed with any type. Inferring to `&AnyStruct` / `&AnyResource` (or even `&Any`) might not be sufficient, for two reasons:
1. The capability could be borrowed with entitlements, but the inferred type does not grant any.

    For example, imagine a capability targeting `/storage/flowTokenVault`, not having a borrow type, and getting borrowed with `auth(FungibleToken.Withdraw) &FlowToken.Vault`. 

2. The value stored under the path could change (e.g. even from a resource to a struct or vice-versa), making the inferred borrow type "too restrictive" 

The first issue is more problematic than the latter. 

Maybe we should just infer a more specific type, based on the stored value at the time of migration. This would make 2) worse, but would at least fix 1)